central/cluster/datastore/datastore_impl.go

```diff
@@ -295,9 +295,10 @@ func (ds *datastoreImpl) Exists(ctx context.Context, id string) (bool, error) {
 func (ds *datastoreImpl) WalkClusters(ctx context.Context, fn func(obj *storage.Cluster) error) error {
 	walkFn := func() error {
 		return ds.clusterStorage.Walk(ctx, func(cluster *storage.Cluster) error {
-			ds.populateHealthInfos(ctx, cluster)
-			ds.updateClusterPriority(cluster)
-			return fn(cluster)
+			clonedCluster := cluster.Clone()
```

Comment:
Sorry can you explain again for dummies why the clone fixes the data race? It seems we can't mutate `cluster` because it may be accessed concurrently. Why is it safe to read for the clone then?

Reply:
The issue wasn't with the read from the walk, but rather that the newly introduced `Walk` function changed _for each reference_ the cluster health status and the priority (which, when read from DB, are apparently not filled necessarily). The `collectCluster`, which showed the data race, was cloning the cluster objects concurrenctly, however the reference got stale due to the `Walk` function updating the underlying health status.

Reply:
This sounds to me like `Walk` itself should already be called with a clone from the store level to avoid this issue.